.github/actions/install-cached-nix/action.yml

```diff
@@ -0,0 +1,37 @@
+name: Install nix with cached nix env
+description: Install nix and attempts to retrieve /nix/store from a cache
+runs:
+  using: "composite"
+  steps:
+  - name: Create /nix/store with liberal permissions
+    run: |
+        # Create with liberal rights, otherwise cache action will complain
+        # about permission errors.
+        sudo mkdir -p /nix/store
+        sudo chmod -R 777 /nix
+    shell: bash
+
+  - name: Cache nix env
+    uses: actions/cache@v2
+    with:
+      path: |
+        # See https://github.com/actions/cache/pull/726
+        /nix/store/**
```

Comment:
I don't recommend trying to cache the nix store with github actions's builtin cache. It's usually better using cachix. If you need to cache nix builds, Tweag's cachix key is available as `TWEAG_CACHIX_AUTH_TOKEN` in the Github secrets.

Reply:
We're pulling everything from IOHK's cache anyway so I'll just skip caching the nix store altogether